enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+Due to the nature of periodic importing, most of the components are sitting idle, especially in the case of Presto which is still utilizing resources despite no queries being processed or evaluated.
+
+With the introduction of Postgres, and the removal of Hive and Presto components, we can continue to utilize the existing Prometheus metrics importer implementation, adapting to the native Postgres data types, and the result is a more streamlined approach to data flow and an overall less complex stack.
+
+This implementation would allow metering to continue using the core, existing CustomResourceDefinitions (CRDs) already exposed in previous releases:
+
+- `ReportDataSource`: exposes an existing table in Postgresql that can be used for reporting needs.
+- `ReportQuery`: defines a SQL query or view that references a `ReportDataSource` table, or a table created by a `Report` object.
+- `Report`: evaluates a templated SQL query defined in an existing `ReportQuery` custom resource and stores those query results in a Hive (now Postgres) table.
+
+At a high-level, the following must be done:
+
+- Create a `PostgresTable` CRD, which is an abstraction that represents an individual table in Postgres.
+- Remove (i.e. avoid reconciling) the internal `HiveTable` and `PrestoTable` custom resources.
+- Update the existing `ReportDataSource` promQL queries to reference the recording rules defined in the metering rule group.
+- Update the existing `ReportQuery` SQL queries and views to index Prometheus labels stored as `jsonb` objects in Postgres tables.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Interacting with Postgres data types
+
+In order to conform with native Postgres data types, any Prometheus labels returned by the query_range API endpoint will need to be represented as the `jsonb` type. This could result in the following `PostgresTable` custom resource instantiation, where metric data is partitioned by datetime and the resultant table column name and types are explicitly defined:
+
+```yaml
+apiVersion: metering.openshift.io/v1
+kind: PostgresTable
+metadata:
+  name: reportdatasource-metering-node-allocatable-cpu-cores
+spec:
+  columns:
+  - name: amount
+    type: float8
+  - name: timestamp
+    type: timestamptz
+  - name: timeprecision
+    type: float8
+  - name: labels
+    type: jsonb
```

Comment:
It's worth noting that Presto had this as its own type, a `map(string, string)` which is probably significantly more efficient to query, but I still don't think is a big deal in the scheme of the typical Metering query.

Reply:
Would it be reasonable to create a lookup table ourselves to implement a `map(string, string)` type functionality that we're used to seeing for tables in Presto, or is translating everything to JSON and using the `jsonb` operator to avoid object mapping?

Reply:
Well, I guess in the latter case, you wouldn't necessarily need to avoid object mapping and you could still use that kind of functionality when interacting with JSON data.

Reply:
Just to close the loop on an out-of-band conversation @timflannagan1 and I had on this. Postgresql can make very efficient use of jsonb.